public/pages/DefineDetector/components/Timestamp/Timestamp.tsx

```diff
@@ -18,16 +18,23 @@ import ContentPanel from '../../../../components/ContentPanel/ContentPanel';
 import { AppState } from '../../../../redux/reducers';
 import { getPrioritizedIndices } from '../../../../redux/reducers/opensearch';
 import { FormattedFormRow } from '../../../../components/FormattedFormRow/FormattedFormRow';
-import { sanitizeSearchText } from '../../../utils/helpers';
+import {
+  getDataSourceFromURL,
+  sanitizeSearchText,
+} from '../../../utils/helpers';
 import { getError, isInvalid, required } from '../../../../utils/utils';
 import { DetectorDefinitionFormikValues } from '../../models/interfaces';
+import { useLocation } from 'react-router-dom';
 
 interface TimestampProps {
   formikProps: FormikProps<DetectorDefinitionFormikValues>;
 }
 
 export function Timestamp(props: TimestampProps) {
   const dispatch = useDispatch();
+  const location = useLocation();
+  const neoQueryParams = getDataSourceFromURL(location);
+  const dataSourceId = neoQueryParams.dataSourceId;
```

Comment:
I could make it just online like 
```
  const dataSourceId = getDataSourceFromURL(useLocation()).dataSourceId;
```
but can I leave it as this way? because:
1. is we need to location object elsewhere, we don't need to define it again
2. when MDS is going to have more field in the future, it's more organized to update MDSQueryParams interface and introduce those new fields here